lightningd/channel_control.c

```diff
@@ -1113,6 +1814,230 @@ void channel_replace_update(struct channel *channel, u8 *update TAKES)
 							  channel->channel_update)));
 }
 
+/* Returns an error if load fails */
+static struct command_result *splice_load_channel(struct command *cmd,
+						  struct channel_id *cid,
+						  struct channel **channel)
+{
+	*channel = channel_by_cid(cmd->ld, cid);
+	if (!*channel)
+		return command_fail(cmd, SPLICE_UNKNOWN_CHANNEL,
+				    "Unknown channel %s",
+				    type_to_string(tmpctx, struct channel_id,
+						   cid));
+
+	if (!feature_negotiated(cmd->ld->our_features,
+			        (*channel)->peer->their_features,
+				OPT_SPLICE))
+		return command_fail(cmd, SPLICE_NOT_SUPPORTED,
+				    "splicing not supported");
+
+	if (!(*channel)->owner)
+		return command_fail(cmd, SPLICE_WRONG_OWNER,
+				    "Channel is disconnected");
+
+	if (!streq((*channel)->owner->name, "channeld"))
+		return command_fail(cmd,
+				    SPLICE_WRONG_OWNER,
+				    "Channel hasn't finished connecting or in "
+				    "abnormal owner state %s",
+				    (*channel)->owner->name);
+
+	if ((*channel)->state != CHANNELD_NORMAL)
+		return command_fail(cmd,
+				    SPLICE_INVALID_CHANNEL_STATE,
+				    "Channel needs to be in normal state but "
+				    "is in state %s",
+				    channel_state_name(*channel));
+
+	return NULL;
+}
+
+static struct command_result *json_splice_init(struct command *cmd,
+					       const char *buffer,
+					       const jsmntok_t *obj UNNEEDED,
+					       const jsmntok_t *params)
+{
+	struct channel_id *cid;
+	struct channel *channel;
+	struct command_result *error;
+	struct splice_command *cc;
+	struct wally_psbt *initialpsbt;
+	struct amount_sat *amount;
+	u32 *feerate_per_kw;
+	bool *force_feerate;
+	u8 *msg;
+
+	if (!param(cmd, buffer, params,
+		  p_req("channel_id", param_channel_id, &cid),
```

Comment:
ideally we call splice_load_channel here: `param_channel_for_splice` perhaps?  That way the `check` command does all those checks, too.